Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -63,23 +65,67 @@ - (instancetype)initWithRecipientId:(NSString *)recipientId
     return self;
 }
 
+- (nullable NSString *)profileName
+{
+    @synchronized(self)
+    {
+        return _profileName;
+    }
+}
+
 - (void)setProfileName:(nullable NSString *)profileName
 {
-    _profileName = [profileName ows_stripped];
+    @synchronized(self)
+    {
+        _profileName = [profileName ows_stripped];
+    }
+}
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(nullable NSZone *)zone
+{
+    UserProfile *copy = [[UserProfile alloc] initWithRecipientId:self.recipientId];
```

Comment:
Can we use the `super` implementation in whole or in part? 

The super implementation is a little more forgiving in that we wouldn't have to remember to update this method every time we add a property here or in super.

Reply:
Done.